File: LD26/Assets/Scripts/GameInfo.js
#pragma strict

///////////////////////
//  CONSTANT FIELDS  //
///////////////////////

static var speechSize : int;
static var titleSize : int;
static var terminalSize : int;

///////////////////////
//    INFO FIELDS    //
///////////////////////

static var gameTitle : String = "Morning (working title)";
static var osName : String = "MorNIX";
public static var protagonistName = "neo"; // change me!
static var antagonistName = "morpheus"; // change me!

///////////////////////



var fontSizeAt1080p : int;

private var screenWidth : int;
private var screenHeight : int;

private static var listeners : List.<ScreenSizeChangeListener>;


function Update () {
	UpdateFontSizes();
}

function UpdateFontSizes(){
	if(screenWidth != Screen.width || screenHeight != Screen.height){
		screenWidth = Screen.width;
		screenHeight = Screen.height;
		
		
		speechSize = (Screen.width / 1920.0) * fontSizeAt1080p;
		titleSize = (Screen.width / 1920.0) * fontSizeAt1080p * 2.0;
		terminalSize = (Screen.width / 1920.0) * fontSizeAt1080p * 0.8;
		print("updated font sizes to: speech=" + speechSize +"pt, title=" + titleSize + "pt, terminal=" + terminalSize);
		
		for(var l : ScreenSizeChangeListener in listeners){
			l.OnScreenSizeChanged();
		}
	}
}

static function RegisterScreenSizeChangeListnener(listener : ScreenSizeChangeListener){
	if(!listeners){
		listeners = List.<ScreenSizeChangeListener>();
	}
	
	listeners.Add(listener);
}

File: LD26/Assets/Scripts/GameInfo.cs
using UnityEngine;
using System.Collections.Generic;

public class GameInfo : MonoBehaviour {

	public const string GAME_TITLE = "Morning (working title)";
	public const string OS_NAME = "MorNIX";
	public const string PROTAGONIST_NAME = "neo"; // change me!
	public const string ANTAGONIST_NAME = "morpheus"; // change me!

	public static int SpeechSize { get; set; }
	public static int TitleSize { get; set; }
	private static int terminalSize;
	
	[SerializeField]
	private int
		fontSizeAt1080p;

	private int screenWidth;
	private int screenHeight;
	private static List<ScreenSizeChangeListener> listeners;
	
	// Update is called once per frame
	void Update() {
		if (screenWidth != Screen.width || screenHeight != Screen.height) {
			screenWidth = Screen.width;
			screenHeight = Screen.height;

			UpdateFontSizes();
		}

		if (true) {
			print("test");
		}
	}

	private void UpdateFontSizes() {
		SpeechSize = (int)((Screen.width / 1920.0) * fontSizeAt1080p);
		TitleSize = (int)((Screen.width / 1920.0) * fontSizeAt1080p * 2.0);
		terminalSize = (int)((Screen.width / 1920.0) * fontSizeAt1080p * 0.8);
		print("updated font sizes to: speech=" + SpeechSize + "pt, title=" + TitleSize + "pt, terminal=" + terminalSize);
		
		foreach (ScreenSizeChangeListener l in listeners) {
			l.OnScreenSizeChanged();
		}
	}
	
	public static void RegisterScreenSizeChangeListnener(ScreenSizeChangeListener listener) {
		if (listeners == null) {
			listeners = new List<ScreenSizeChangeListener>();
		}
		
		listeners.Add(listener);
	}
}
